src/v/storage/disk_log_impl.cc

```diff
@@ -801,7 +801,7 @@ ss::future<> disk_log_impl::retention_adjust_timestamps(
       model::timestamp::now().value() + ignore_in_future / 1ms);
 
     for (const auto& s : _segs) {
-        auto max_ts = s->index().max_timestamp();
```

Comment:
I think this is fine, but my initial reaction was that it's nice to have this recomputing because our conditions may change: `now` trudges along forward meaning the ignore boundary changes. It isn't clear that this behavior is worse though, so I'm fine keeping this change in.

Separately, I wonder if we should be recomputing the retention timestamp each time `storage_ignore_timestamps_in_future_sec` changes, though that seems non-trivial and it's unclear whether there's much value in doing so